### ISSUE TITLE
Cursor navigation without text selection

### ISSUE THREAD

**Issue by perlinm** [helix-editor/helix#3114]
#### Describe your feature request

TLDR: Cursor navigation commands (`move_*` and `goto_*`) that do **not** select text.

Most cursor navigation commands** behave as expected in `normal_mode`: they simply move the cursor (without selecting any text).  Some cursor navigation commands***, however, also select the text between the current cursor location and the target cursor location.

Aside from being inconsistent, automatic text selection for some cursor navigation commands leads to unexpected and unintuitive behavior.  Starting from `normal_mode`, if I (1) `goto_next_paragraph` and then (2) enter `insert_mode`,  I find myself editing text at the previous location of the cursor (i.e., at the start of the selected text).  I would like to navigate to the next paragraph without selecting it.

** Example commands that move the cursor without text selection: `move_line_[up/down]`, `move_char_[left/right]`, `goto_line_[start/end]`
*** Commands that automatically select text (non-exhaustive): `move_[next/prev]_word_start`, `goto_[next/prev]_paragraph`

**Comment by archseer** [comment]
> Aside from being inconsistent

It's perfectly consistent with kakoune :) Some commands don't select text because they're jumps, others do because they're select based movements.

> I find myself editing text at the previous location of the cursor (i.e., at the start of the selected text)

Yeah insert edits at the start of the selection, append at the end of the selection.

If you'd like to avoid selections you can remap commands to command + collapse_selection, but it will make certain edits require more keypresses, eg. wd -> vwd

**Comment by perlinm** [comment]
I still consider this behavior strange, but command + collapse_selection will do the job.  Thanks :slightly_smiling_face:

**Comment by the-mikedavis** [comment]
> I would personally urge helix developers to reconsider mimicking the behavior of other text editors in favor of thinking more deeply about what is "reasonable" behavior

You're urging us to consider mimicking vim instead of kakoune though. Motions creating/extending selections is reasonable and useful for an editor that implements the select-then-act paradigm: every action acts on a selection so motions should create selections. Select-then-act is deeply embedded into helix and will not be changed.

> Cursor navigation and text selection are conceptually distinct primitives, and should be treated as such

That they are not distinct is a feature: all motions and jumps create, move, or extend selections. The cursor itself is a selection with a width of at least one (hence why `d` deletes the current character under the cursor when the cursor is only over one character).

Motions creating selections can be a bit distracting at first but I would urge you to try it for a while and see if you can adjust - you may end up liking it more. If not, (n)vim's verb-object paradigm might be a better fit for you.

**Comment by perlinm** [comment]
I did not ask for mimicking any other editor: notice that I only referred to existing helix commands such as `move_char_right`, `move_line_down`, and `goto_line_end`.  This is what I mean by being inconsistent: the behavior of `move_line_down` and `move_next_word_start` is **qualitatively different** in a way that the names of these commands do not reflect.

Maybe I should have emphasized the inconsistency rather than behavior in the title of this request, but admittedly since I have figured out how to get the behavior I want, the rest is aesthetic.

**Comment by the-mikedavis** [comment]
> Helix already has text navigation without cursor selection

No, this is incorrect. All motions including `move_line_down` create selections. Check out the source for `move_impl` and `move_word_impl` in `helix-term/src/commands.rs` - the implementations are nearly identical.

**Comment by perlinm** [comment]
Sorry, I meant to say that "Helix already has cursor navigation without text selection" (not counting the trivial one-character selection): `move_line_down` only **moves** the cursor to a new location, whereas `move_next_word_start` makes a new multi-character selection.

**Comment by vincentole** [comment]
I have the same issue and I find myself to almost always type `;` before entering insert mode. 
I will try to remap i to the following:
`i = ["collapse_selection", "insert_mode"]`

If I want to insert before selection I can still type `A-;` to put the cursor to the start of the selection.
Maybe this helps someone :)
